dts/arm/ambiq/ambiq_apollo4p.dtsi

```diff
@@ -96,9 +96,9 @@
 			status = "okay";
 		};
 
-		counter0: counter@40008000 {
+		counter0: counter@40008200 {
 			compatible = "ambiq,counter";
-			reg = <0x40008000 0x80>;
+			reg = <0x40008200 0x20>;
```

Comment:
I think the register base address should be kept the same with the chip hardware definition.

Reply:
The Apollo4 counter may have to differ since the underlying resigsters are not identical.

I have an Ambiq branch with fixed to the apollo4 counter. Let's sync up offline.

Reply:
Understood the previous register address here is incorrect and not used by the driver. Yes we should correct them in the DTS.

Reply:
Yes, the previous driver fixed the counter instance to 0 and the register address defined in dts had not been used. As I added support for multi-instance, the base address and offset should be changed to the correct ones.

Reply:
I already had a branch with changes in it to fix this. I will check against my branch before approving.